data/account.js

```diff
@@ -61,4 +63,58 @@ module.exports = {
     const newId = insertInfo.insertedId.toString();
     return newId;
   },
-};
+
+  async login(username, password) {
+    if (!username) throw "must provide username"
+    if (!password) throw "must provide password"
+
+    if (typeof username !== "string")
+        throw "Error: username should be a string";
+    if (typeof password !== "string")
+        throw "Error: password should be a string";
+
+        //comment 
+    if (username.indexOf(" ") >= 0)
+        throw "Error: username should not have any spaces";
+    if (password.indexOf(" ") >= 0)
+        throw "Error: password should not have any spaces";
+
+
+    username = username.trim();
+    password = password.trim();
+
+    if (username.length < 2)
+        throw "Error: username must have at least two characters";
+    if (password.length < 6)
+        throw "Error: password must have at least eight characters";
+
+        
+    //check for alphnumeric 
+    //https://stackoverflow.com/questions/4434076/best-way-to-alphanumeric-check-in-javascript
+
+    for (let i = 0; i < username.length; i++) {
```

Comment:
I think in the future you can replace this with a regex to make the code simpler